packages/palette/src/elements/Toggle/Toggle.tsx

```diff
@@ -0,0 +1,100 @@
+import React from "react"
+import styled, { css } from "styled-components"
+import { Box, BoxProps } from "../Box"
+import { SWITCH_STATES, TOGGLE_STATES } from "./tokens"
+import { themeGet } from "@styled-system/theme-get"
+
+export interface ToggleProps
+  extends BoxProps,
+    Omit<React.HTMLAttributes<HTMLDivElement>, "onSelect"> {
+  /** Disable toggle interactions */
+  disabled?: boolean
+  /** Select the toggle on render */
+  selected?: boolean
+  /** Used to force the toggle into the visual hover state */
+  hover?: boolean
+  /** Callback when selected */
+  onSelect?: (selected: boolean) => void
+}
+
+/** A toggle */
+export const Toggle: React.FC<ToggleProps> = ({
+  selected = false,
+  disabled,
+  onSelect,
+  onClick,
+  ...rest
+}) => {
+  const isSelectable = !disabled && onSelect !== undefined
+
+  const handleClick = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+    if (isSelectable && onSelect) {
+      onSelect(!selected)
+    }
+
+    if (onClick) {
+      onClick(event)
+    }
+  }
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    if (event.key === " " && isSelectable && onSelect) {
+      event.preventDefault()
+      onSelect(!selected)
+    }
+  }
+
+  return (
+    <Container
+      display="flex"
+      alignItems="center"
+      onClick={handleClick}
+      tabIndex={disabled ? -1 : 0}
+      onKeyPress={handleKeyPress}
+      role="toggle"
```

Comment:
I believe this should be `role="switch"` https://www.w3.org/WAI/ARIA/apg/patterns/switch/